resources/todomvc/architecture-examples/backbone/scripts/build.js

```diff
@@ -0,0 +1,56 @@
+const fs = require("fs").promises;
+
+const rootDirectory = "./";
+const sourceDirectory = "./src";
+const targetDirectory = "./dist";
+
+const htmlFile = "index.html";
+
+const filesToMove = ["node_modules/todomvc-common/base.css", "node_modules/todomvc-app-css/index.css", "node_modules/jquery/dist/jquery.min.js", "node_modules/underscore/underscore-min.js", "node_modules/backbone/backbone-min.js"];
+
+const copy = async (src, dest) => {
+    await fs.copyFile(src, dest);
+};
+
+const build = async () => {
+    // remove dist directory if it exists
+    await fs.rm(targetDirectory, { recursive: true, force: true });
+
+    // re-create the directory.
+    await fs.mkdir(targetDirectory);
+
+    // copy src folder
+    await fs.cp(sourceDirectory, targetDirectory, { recursive: true }, (err) => {
+        if (err) console.error(err);
+    });
+
+    // copy html file
+    await fs.copyFile(`${rootDirectory}${htmlFile}`, `${targetDirectory}/${htmlFile}`);
+
+    // copy files to move
+    for (let i = 0; i < filesToMove.length; i++) {
+        const fileName = filesToMove[i].split("/").pop();
+        await copy(filesToMove[i], `${targetDirectory}/${fileName}`);
+    }
+
+    // read html file
+    let html = await fs.readFile(`${targetDirectory}/${htmlFile}`, "utf8");
+
+    // remove base paths from files to move
+    for (let i = 0; i < filesToMove.length; i++) {
+        const basePath = `${filesToMove[i].split("/").slice(0, -1).join("/")}/`;
```

Comment:
It seems like something where we replaced i.e. `"node_modules/jquery/dist/jquery.min.js"` with `"jquery.min.js"` would be better than replacing `"node_modules/jquery/dist/"` with `""`. Haven't worked it all the way out but I feel like there could be a bug with subsequent paths in nested subdirectories otherwise. Something like this perhaps (instead of calculating basePath)

```
html = html.replace(filesToMove[i], filesToMove[i].slice(-1))
```

Reply:
we could re-use line 33 and update it to:

```
const fileName = filesToMove[i].split("/").pop();
html = html.replace(filesToMove[i], fileName);
```

Reply:
yeah something like that seems fine

Reply:
done